ocaml/xapi/nm.ml

```diff
@@ -468,12 +468,25 @@ let bring_pif_up ~__context ?(management_interface=false) (pif: API.ref_PIF) =
       end;
 
       (* sync MTU *)
+      begin
       try
         let mtu = Int64.of_int (Net.Interface.get_mtu dbg ~name:bridge) in
         if mtu <> rc.API.pIF_MTU then
           Db.PIF.set_MTU ~__context ~self:pif ~value:mtu
       with _ ->
         warn "could not update MTU field on PIF %s" rc.API.pIF_uuid
+      end;
+      
+      (* sync igmp_snooping_enabled *)
+      if (rc.API.pIF_VLAN = -1L) && (rc.API.pIF_bond_slave_of = Ref.null) && (rc.API.pIF_tunnel_access_PIF_of == []) then begin
+        let igmp_snooping = Db.Pool.get_igmp_snooping_enabled ~__context ~self:(Helpers.get_pool ~__context) in
+        let igmp_snooping' = match igmp_snooping with
```

Comment:
A match on a Boolean is a little odd. Just do

    if igmp_snooping then `True else `False

(or `enabled` and `disabled`)